src/access/ownable/ownable.cairo

```diff
@@ -30,20 +33,68 @@ mod Ownable {
         const ZERO_ADDRESS_OWNER: felt252 = 'New owner is the zero address';
     }
 
+    #[embeddable_as(OwnableImpl)]
+    impl Ownable<
+        TContractState, +HasComponent<TContractState>
+    > of interface::IOwnable<ComponentState<TContractState>> {
+        /// Returns the address of the current owner.
+        fn owner(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Ownable_owner.read()
+        }
+
+        /// Transfers ownership of the contract to a new address.
+        fn transfer_ownership(
+            ref self: ComponentState<TContractState>, new_owner: ContractAddress
+        ) {
+            assert(!new_owner.is_zero(), Errors::ZERO_ADDRESS_OWNER);
+            self.assert_only_owner();
+            self._transfer_ownership(new_owner);
+        }
+
+        /// Leaves the contract without owner. It will not be possible to call `assert_only_owner`
+        /// functions anymore. Can only be called by the current owner.
+        fn renounce_ownership(ref self: ComponentState<TContractState>) {
+            self.assert_only_owner();
+            self._transfer_ownership(Zeroable::zero());
+        }
+    }
+
+    #[embeddable_as(OwnableCamelOnlyImpl)]
+    impl OwnableCamelOnly<
+        TContractState, +HasComponent<TContractState>
+    > of interface::IOwnableCamelOnly<ComponentState<TContractState>> {
+        fn transferOwnership(ref self: ComponentState<TContractState>, newOwner: ContractAddress) {
+            self.transfer_ownership(newOwner);
+        }
+
+        fn renounceOwnership(ref self: ComponentState<TContractState>) {
+            self.renounce_ownership();
+        }
+    }
```

Comment:
I see your point, but the issue I have with it is that I think users should have the ability to opt-in/out of some impl for components, and having a single big ABI enforces every function by default. Examples of components that I think should have multiple external impl are:

ERC721: the ERC721Metadata impl should be optional, even when is widely used. The standard doesn't require this, and users should be able to not use it IMO.

The ERC20 example as well should have the metadata part as optional, but also the non-standard impl IMO.

I think I lean towards letting multiple impl, even when I feel (agree) this could be improved somehow.

Reply:
> I see your point, but the issue I have with it is that I think users should have the ability to opt-in/out of some impl for components, and having a single big ABI enforces every function by default.

Oh, for sure! I was thinking more along the lines of having the implementations that we have now and also one big ABI impl with everything. This gives the user the option to either choose specific impls or use the whole component. I imagine this being quite beneficial for the account and erc721 components

The issue I see, though, is we'd have to rewrite all the methods under the ABI impl or create some sort of multi-impl embeddable mechanism

Reply:
I lean towards modularity here, having ERC721, ERC721Metadata, and non-standard (with a more descriptive name, not generic) impls separately and required explicitly in the contract. Having an extra impl with the full ABI seems like an antipattern, because it would be harder to follow IMO.